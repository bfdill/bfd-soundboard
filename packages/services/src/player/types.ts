import { ChildProcess } from 'child_process'

export type DefaultPlayer =
  | 'mplayer'
  | 'afplay'
  | 'mpg123'
  | 'mpg321'
  | 'play'
  | 'omxplayer'
  | 'aplay'
  | 'cmdmp3'
export type What = string | string[]
export type PlayerOptions = {
  player: any
  players?: DefaultPlayer | DefaultPlayer[] | string[]
}
export type PlayerArgs = { [playerName: string]: string[] }
export type PlayOptions = PlayerArgs & { stdio: 'ignore' }
export type Next = Function
export interface IPlayer {
  play: (what: What, options?: PlayOptions | Next, next?: Next) => ChildProcess
}
export type createPlayer = (opts?: PlayerOptions) => IPlayer
