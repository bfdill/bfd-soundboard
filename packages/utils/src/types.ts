export type DeepPartial<T> = {
  [P in keyof T]?: T[P] extends (infer U)[]
    ? DeepPartial<U>[]
    : T[P] extends ReadonlyArray<infer U>
    ? ReadonlyArray<DeepPartial<U>>
    : T[P] extends (string | number | boolean)
    ? T[P]
    : DeepPartial<T[P]>
}
