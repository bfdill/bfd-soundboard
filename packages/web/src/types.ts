import { Action } from 'redux'
import { ThunkAction } from 'redux-thunk'

export type FullState = {
  sounds: any
}

export type RequestGetSoundsAction = Action<'REQUEST/getSounds'>
export type SuccessGetSoundsAction = Action<'SUCCESS/getSounds'> & {
  payload: any
}
export type FailGetSoundsAction = Action<'FAILED/getSounds'> & { err: Error }
export type AnyAction =
  | RequestGetSoundsAction
  | SuccessGetSoundsAction
  | FailGetSoundsAction
export type GetSounds = () => ThunkAction<
  void,
  FullState,
  undefined,
  RequestGetSoundsAction | SuccessGetSoundsAction | FailGetSoundsAction
>
