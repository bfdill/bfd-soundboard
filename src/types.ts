import * as Koa from 'koa'
import * as KoaRouter from 'koa-router'
import { Server } from 'http'

export interface IApplication {
  listen: (port: number) => Server
}

export type ApplicationState = {
  correlationId: string
}
export interface IApplicationContext
  extends Koa.ParameterizedContext<ApplicationState> {}
export type ApplicationMiddleware = Koa.Middleware<
  ApplicationState,
  IApplicationContext
>
export type ApplicationKoa = Koa<ApplicationState, IApplicationContext>
export type ApplicationRouter = KoaRouter<ApplicationState, IApplicationContext>
