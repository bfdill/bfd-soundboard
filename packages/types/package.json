{
  "name": "@talkboy-9000/types",
  "version": "0.0.1"
}
