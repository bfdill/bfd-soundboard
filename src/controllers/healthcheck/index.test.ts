import {
  getHealthcheckRouter,
  HealthcheckController,
  IHealthcheckController,
  getHealthcheckControllerLogger
} from '.'
import { IApplicationContext, ApplicationState } from '../../types'
import { OK } from 'http-status-codes'
import {
  getJestLogger,
  IJestLogger,
  snapshotExistingLogger
} from '../../modules/winston-jest/index.test'

describe('controllers -> healthcheck', () => {
  const jestLogger: IJestLogger = getJestLogger()
  const healthcheck: IHealthcheckController = new HealthcheckController(
    jestLogger.logger
  )
  const testState: ApplicationState = { correlationId: 'abc123' }
  const testContext: IApplicationContext = { state: testState } as any

  test('has known exports', () => {
    expect(Object.keys(require('.'))).toMatchSnapshot()
  })

  test('getHealthcheckLogger', () => {
    snapshotExistingLogger(getHealthcheckControllerLogger())
  })

  test('singleton works', () => {
    const expected = HealthcheckController.getInstance()
    const actual = HealthcheckController.getInstance()

    expect(actual).toBe(expected)
  })

  test('healthy, so healthy!', async () => {
    await healthcheck.health(testContext)

    expect(testContext.body).toEqual({ healthy: true })
    expect(testContext.status).toEqual(OK)
    jestLogger.callsMatchSnapshot()
  })

  test('getHealthcheckRouter', () => {
    expect(getHealthcheckRouter()).toMatchSnapshot()
  })
})
