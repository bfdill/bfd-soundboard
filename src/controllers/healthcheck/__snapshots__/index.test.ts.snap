// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controllers -> healthcheck getHealthcheckRouter 1`] = `
Router {
  "methods": Array [
    "HEAD",
    "OPTIONS",
    "GET",
    "PUT",
    "PATCH",
    "POST",
    "DELETE",
  ],
  "opts": Object {},
  "params": Object {},
  "stack": Array [
    Layer {
      "methods": Array [
        "HEAD",
        "GET",
      ],
      "name": null,
      "opts": Object {
        "end": true,
        "ignoreCaptures": undefined,
        "name": null,
        "prefix": "",
        "sensitive": false,
        "strict": false,
      },
      "paramNames": Array [],
      "path": "/",
      "regexp": /\\^\\(\\?:\\\\/\\(\\?=\\$\\)\\)\\?\\$/i,
      "stack": Array [
        [Function],
      ],
    },
  ],
}
`;

exports[`controllers -> healthcheck has known exports 1`] = `
Array [
  "HealthcheckController",
  "getHealthcheckRouter",
]
`;

exports[`controllers -> healthcheck healthy, so healthy! 1`] = `
Array [
  Object {
    "controller": Object {
      "name": "HealthcheckController",
    },
    "correlationId": "default_mock_correlation_id",
    "level": "silly",
    "message": "success",
    Symbol(level): "silly",
    Symbol(splat): Array [
      Object {
        "correlationId": "default_mock_correlation_id",
      },
    ],
    Symbol(message): "{\\"controller\\":{\\"name\\":\\"HealthcheckController\\"},\\"correlationId\\":\\"default_mock_correlation_id\\",\\"level\\":\\"silly\\",\\"message\\":\\"success\\"}",
  },
]
`;
